### ISSUE TITLE
Next deployment (after 2023-06-28) to openflights.org

### ISSUE THREAD

**Issue by reedy** [jpatokal/openflights#1391]
Relevant tasks/PR's that may require action for the next deployment to openflights.org.

I believe on 2023-06-28 it was updated to ecf635a4bc460a5ce3a911c34305672188ad6d80.

Changes: https://github.com/jpatokal/openflights/compare/ecf635a4bc460a5ce3a911c34305672188ad6d80...master

After #1216 / a70afa112267e4f0f21bd3ef516f690958facf93 this will be easier to check in future from the live site.

- [x] #1388 - `php7.4-intl` needs installing (and then nginx/fpm restarts as appropriate)
- [x] #1232 / 9108498f02e6caaeab37da8ab3209e4daab47e35 - Update `config.php` to replace `$OF_ADMIN_UID = 3;` with `$OF_ADMIN_UID = [ 3 ];`
- [x] #1360 / 7a6ca62904e334862c8e54dfc1f96f753069ae0b - Update `config.php` to add `$OF_ENABLE_SIGNUP = true;`
- [x] #1232 / 968137cc25516a555d12a467a7cb1ab53e417b8f - Update `config.php` to add `$OF_DEMO_UID = 1;`
- [x] #1355 / 071cd3ec186c2118b7c780197aa07ffd16523611 - Remove `$OF_VBULLETIN_LOGIN` from config.php
- [ ] #1314 / 5e0fae7fa7d44e4414d3ea28b585da1bc5c2712b - Check and update `$GITHUB_ACCESS_TOKEN` if it's expired (it might not be, just the previous code update was rubbish in c03f72006444dfc2485eec28ba936d3d74d1631d).
  - May want to check the submission works afterwards too.
  - It might be necessary/useful to do #1171 too; add/create an `openflights-bot` user (or similar), and create a token on that account.
  - Maybe leave a comment in the `config.php` when the token is due to expire; as the newer token style only last a year.

TODOs:
- [x] #1266
- [ ] Add `en_GB` locale to `locale` table
  - #1404 - Make updating `locale` table easier; don't just want to run `sql/load-data.sql`. Split locale stuff into a new/separate/extra file so it can be run separately?
- [ ] Do we need to do any locale related regeneration server side? (Need more documentation here...)
- [ ] #1414
  - [ ] Update `route` table. [Data updated `2023-03-22`](http://arm.64hosts.com/).
    - Add `schedule` table schema to #1408
    - Run `tools/update-arm.sh`
    - Run `sql/import-arm.sql`
    - Export updates routes to `data/routes.dat`
 - [ ] #1435

---

Other fixes (missing files/crons):

- [ ] #1136 - Run `sql/top10.sql` and add/fix cron job
- [ ] #1225 - Run `sql/update-demo.sql` and add/fix cron job

---

Errors from logs that could do with investigating (or at least, logs posting on tasks):
- [ ] #1314 - Possibly fixed with https://github.com/jpatokal/openflights/commit/5e0fae7fa7d44e4414d3ea28b585da1bc5c2712b and necessary token updates mentioned above
- [x] #1178

**Comment by jpatokal** [comment]
@reedy Are you OK with me doing a deploy from head to prod?

**Comment by reedy** [comment]
> @reedy Are you OK with me doing a deploy from head to prod?

I believe so, yeah.

I'm not going to say 100% everything is going to work right, but...

I think the big things that need doing before upgrading, are as listed in the task description. Just to prevent things failing spectacularly.

Install `php7.4-intl` and restart service (fpm/nginx) as appropriate.

As long as you update `config.php` as appropriate (ideally adding `$OF_ENABLE_SIGNUP` and `$OF_DEMO_UID`; though the former doesn't have to be added, due to checks in the code, but it'd be nice to remove the `isset()` checks. Also updating `$OF_ADMIN_UID` after too).

Happy to try and be around when you do it, if you want. I keep a weird local TZ.

**Comment by jpatokal** [comment]
OK, thanks.  This sounds involved enough that I'm not going to risk it without access to a proper computer if things go pear-shaped, so I'll put this on hold for two weeks or so.

**Comment by jpatokal** [comment]
Apologies for the massive delay.  I have updated the configs, pulled from head, rebooted nginx and done a quick sanity check, and the site seems to work.  Only bump in the road: I had to explicitly install `php7.4-mbstring` as well, not just `php7.4-intl`.  I've followed up on #1314 re: the Github token, AFAICT it has no expiration date.

**Comment by reedy** [comment]
Bleugh, that's my fault - 39226ef13e6982726f352df87ddf6bb4e653b2a3 for #1412

Will put it in `composer.json`.

I honestly can't remember if I already had it installed in my dev vm for one reason ro another...

**Comment by reedy** [comment]
Hmm, it was already listed in `composer.json` since before a728c1deaa92c5b2b78c61e17278eafb7945b019 (which suggests there was a `mb_` call somewhere..).

No big deal at least, easily enough resolved :)

**Comment by reedy** [comment]
Hmm. First load of the site (existing browser or incognito) I see

![Screenshot 2023-10-19 at 11 03 20](https://github.com/jpatokal/openflights/assets/67615/d3cb7f16-1479-496d-845c-7159a353b0b7)

Click OK a couple of times, change the locale and we're ok...

Some something to follow up on there

**Comment by jpatokal** [comment]
Works for me(tm) in incognito.  Might be a locale issue, which one are you using?  `en_GB`?

**Comment by reedy** [comment]
It will be trying to use `en_GB` yeah, but I notice it's not in the drop down (because it won't be in the `locales` table, which is another issue, and mentioned in the task :))...

Browser console tells all:

![Screenshot 2023-10-19 at 12 25 49](https://github.com/jpatokal/openflights/assets/67615/5a2ddb97-5718-42e5-9bba-670b9ba6078f)

Looks like a fallback isn't quite working; and/or the check for `null` needs to be checking for `''` too.

**Comment by reedy** [comment]
5e3957f41a76da365d9b0aad2bd03f719959c799 and b9b58a706a1cbf4fde66480552046e717f3aa849 should take care of this!

**Comment by jpatokal** [comment]
I've done another pull from head to fix the locale issues, which were causing significant pain (#1448).

**Comment by jpatokal** [comment]
@reedy I suggest we close this out and open a new issue for the next bigger release.
